### DISCUSSION TITLE
Redraw embedded opengl texture without redrawing entire ui

### DISCUSSION THREAD

**Discussion by AlexanderHarrison** [slint-ui/slint#3190] [Q&A] [answered]
Hi, I've embedded an opengl texture that needs to redraw every frame inside of a slint application. It seems that calling `app.window().request_redraw()` every frame redraws the entire ui, and if not called then the updated texture is not shown until the next ui redraw. Is there a method to only redraw that texture every frame? The texture is embedded in an `Image`, exactly like the `opengl_texture` example.

I have considered using an opengl overlay and testing if I can redraw the overlay without redrawing the ui underneath, but I figured I should ask for input before attempting this.

**Comment by tronical** [comment, marked as answer]
I think what you're doing is correct, calling `request_redraw()`. It's true that right now ever frame with OpenGL is rendered from scratch. There are two extensions that we could use to improve on this (buffer age and swap with damage), and we should. When that's implemented, `request_redraw()` will automatically only redraw what changed (your texture).
